BlueprintUICommonControls/Sources/Internal/KeyboardObserver.swift

```diff
@@ -0,0 +1,172 @@
+//
+//  KeyboardObserver.swift
+//  BlueprintUICommonControls
+//
+//  Created by Kyle Van Essen on 2/16/20.
+//
+
+import UIKit
+
+
+protocol KeyboardObserverDelegate : AnyObject
+{
+    func keyboardFrameWillChange(observer : KeyboardObserver)
+}
+
+/**
+ Implementation borrowed from Listable:
+ https://github.com/kyleve/Listable/blob/master/Listable/Sources/Internal/KeyboardObserver.swift
+ 
+ iOS Docs for keyboard management:
+ https://developer.apple.com/library/archive/documentation/StringsTextFonts/Conceptual/TextAndWebiPhoneOS/KeyboardManagement/KeyboardManagement.html
+ */
+final class KeyboardObserver
```

Comment:
Why not build this into the scroll view? It seems too tightly coupled to be reusable with any other view.

Reply:
I don't mind keeping it separate – it's a chunk of logic that's pretty easy to contain. I wish `UIKit` exposed an API more like this, to be honest.

Reply:
Yeah, wanted to keep this separate because the nuts and bolts of it aren't related to UIScrollView at all – could see us using this in more elements, plus it makes it easier to test.